### PR TITLE
Add unitSpawnEvent

### PR DESCRIPTION
**Pull request by Darkness6030**
Mindustry doesn't have an event which is triggered when a unit is spawned by wave. 
So i've added it

### CODE REVIEW

File: core/src/mindustry/game/EventType.java [hunk 1]
```diff
@@ -428,6 +428,15 @@ public UnitCreateEvent(Unit unit, Building spawner){
         }
     }
 
+    /**Called when a unit is spawned by wave. */
```
Comment (buthed010203) [inline core/src/mindustry/game/EventType.java:431]: You forgot a space after /**

File: core/src/mindustry/game/EventType.java [hunk 2]
```diff
@@ -428,6 +428,15 @@ public UnitCreateEvent(Unit unit, Building spawner){
...
+    public static class UnitSpawnEvent{
+        public final Unit unit;
+
+        public UnitSpawnEvent(Unit unit) {
```
Comment (buthed010203) [inline core/src/mindustry/game/EventType.java:435]: No space before {

**Review by buthed010203 — CHANGES_REQUESTED** [review]

File: core/src/mindustry/game/SpawnGroup.java [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 import arc.struct.*;
 import arc.util.*;
+import arc.Events;
```
Comment (MEEPofFaith) [inline core/src/mindustry/game/SpawnGroup.java:5]:
```suggestion
import arc.*;
```

Use wildcard imports.

**Comment by MEEPofFaith** [comment]
Also where PR template checkboxes?

**Comment by buthed010203** [comment]
Also how does this differ from unitCreateEvent or whatever its called?

**Comment by Darkness6030** [comment]
@buthed010203, UnitCreateEvent is called when unit is created by a factory
UnitSpawnEvent is called when a unit is spawned by spawnPos

**Pull request merged**